### PR TITLE
[techsupport] Handle minor fixes of TS Lock and update auto-TS

### PR DESCRIPTION
**Pull request by vivekreddynv**
Signed-off-by: Vivek Reddy Karri <vkarri@nvidia.com>

#### What I did

1) Print the last statement as the techsupport dump name, as some automation processes might depend of parsing the last line to infer the dump path.

```
Previously:
handle_exit
Removing lock. Exit: 0
removed '/tmp/techsupport-lock/PID'
removed directory '/tmp/techsupport-lock'

Updated:
handle_exit
Removing lock. Exit: 0
removed '/tmp/techsupport-lock/PID'
removed directory '/tmp/techsupport-lock'
/var/dump/sonic_dump_r-bulldog-03_20220324_195553.tar.gz
```

2) Don't acquire the lock when running in NOOP mode

3) Set the set -v option just before running main so that it won't print the generate_dump code to stdout

4) Update the auto-techsupport script to handle EXT_RETRY and EXT_LOCKFAIL exit codes returned by show techsupport command. 

5) Update the minor error in since argument for auto-techsupport

### CODE REVIEW

**Review by dgsudharsan — APPROVED** [review]

File: scripts/coredump_gen_handler.py [hunk 1]
```diff
@@ -110,13 +110,22 @@ def parse_ts_dump_name(self, ts_stdout):
...
         cmd_opts = ["show", "techsupport", "--silent", "--since", since_cfg]
         cmd  = " ".join(cmd_opts)
         rc, stdout, stderr = subprocess_exec(cmd_opts, env=ENV_VAR)
-        if rc:
+        if rc == EXT_LOCKFAIL:
+            syslog.syslog(syslog.LOG_NOTICE, "Another instance of techsupport running, aborting this. stderr: {}".format(stderr))
+            return ""
+        elif rc == EXT_RETRY:
+            if num_retry <= MAX_RETRY_LIMIT:
+                print(num_retry)
```
Comment (dgsudharsan) [inline scripts/coredump_gen_handler.py:122]: Should there be a gap of few seconds before next retry?

File: utilities_common/auto_techsupport_helper.py [hunk 2]
```diff
@@ -60,6 +61,11 @@
 TIME_BUF = 20
 SINCE_DEFAULT = "2 days ago"
 
+# Techsupport Exit Codes
+EXT_LOCKFAIL = 2
+EXT_RETRY = 4
+EXT_SUCCESS = 0
+MAX_RETRY_LIMIT = 2
```
Comment (dgsudharsan) [inline utilities_common/auto_techsupport_helper.py:68]: Is this configurable?

File: scripts/coredump_gen_handler.py (hunk 1 above)
Comment (vivekreddynv) [reply scripts/coredump_gen_handler.py:122]: Not required, EXT_RETRY happening is less likely and response should be quick in order to grab the lock. so i don't think a gap is required. 
I'll remove the print statement though

**Review by dgsudharsan — DISMISSED** [review]

**Comment by dgsudharsan** [comment]
@qiluo-msft  Can you please help to review this PR?

File: scripts/coredump_gen_handler.py [hunk 3]
```diff
@@ -110,13 +110,21 @@ def parse_ts_dump_name(self, ts_stdout):
...
+        if rc == EXT_LOCKFAIL:
+            syslog.syslog(syslog.LOG_NOTICE, "Another instance of techsupport running, aborting this. stderr: {}".format(stderr))
+            return ""
+        elif rc == EXT_RETRY:
+            if num_retry <= MAX_RETRY_LIMIT:
+                return self.invoke_ts_cmd(since_cfg, num_retry+1)
+            else:
+                syslog.syslog(syslog.LOG_ERR, "MAX_RETRY_LIMIT for show techsupport invocation exceeded, stderr: {}".format(stderr))
+        elif rc != EXT_SUCCESS:
             syslog.syslog(syslog.LOG_ERR, "show techsupport failed with exit code {}, stderr: {}".format(rc, stderr))
```
Comment (qiluo-msft) [inline scripts/coredump_gen_handler.py:126]: >syslog [](http://example.com/codeflow?start=19&length=6)

Do you want to exit immediately after the error logging?

File: scripts/generate_dump [hunk 4]
```diff
@@ -63,6 +63,10 @@ handle_exit()
     ECODE=$?
     echo "Removing lock. Exit: $ECODE" >&2
     $RM $V -rf ${LOCKDIR}
+    # Echo the filename as the last statement if the generation suceeds
```
Comment (qiluo-msft) [inline scripts/generate_dump:66]: >suceeds [](http://example.com/codeflow?start=64&length=7)

typo

Comment (qiluo-msft) [inline scripts/generate_dump:66]: >Echo [](http://example.com/codeflow?start=6&length=4)

Isn't it already supported? I am seeing
```
mkdir: created directory '/var/dump/sonic_dump_vlab-01_20220331_204737/log'
sonic_dump_vlab-01_20220331_204737/log/techsupport_time_info.Gg18UE4AVH
removed '/var/dump/sonic_dump_vlab-01_20220331_204737/log/techsupport_time_info.Gg18UE4AVH'
removed directory '/var/dump/sonic_dump_vlab-01_20220331_204737/core'
removed directory '/var/dump/sonic_dump_vlab-01_20220331_204737/log'
removed directory '/var/dump/sonic_dump_vlab-01_20220331_204737'
/var/dump/sonic_dump_vlab-01_20220331_204737.tar:         5.5% -- replaced with /var/dump/sonic_dump_vlab-01_20220331_204737.tar.gz
/var/dump/sonic_dump_vlab-01_20220331_204737.tar.gz
```

**Review by ganglyu — APPROVED** [review]

File: scripts/generate_dump (hunk 4 above)
Comment (vivekreddynv) [reply scripts/generate_dump:66]: After the addition of lock code, no. handle_exit is the last trap that runs before exiting, and it does print a few statements and thus the issue

File: scripts/coredump_gen_handler.py (hunk 3 above)
Comment (vivekreddynv) [reply scripts/coredump_gen_handler.py:126]: yeah, i'll add it

**Review by qiluo-msft — APPROVED** [review]
LGTM

**Comment by liat-grozovik** [comment]
@ganglyu could yuo please review recent changes following feedback provided?

**Review by ganglyu — APPROVED** [review]

**Pull request merged**